examples/silentpayments.c

```diff
@@ -0,0 +1,503 @@
+                );
+                secp256k1_ec_pubkey_serialize(ctx,
+                    labels_cache.entries[0].label,
+                    &len,
+                    &label,
+                    SECP256K1_EC_COMPRESSED
+                );
+                assert(ret);
+                labels_cache.entries_used = 1;
+            }
+
+            /* Bob collects the public data from the transaction inputs and
+             * creates a `secp256k1_silentpayments_public_data` object. He uses
+             * this for his own scanning and also serializes the `public_data`
+             * object to send to light clients. We will use this later for
+             * Carol, who is scanning as a light client. Note, anyone can create
+             * and provide these `public_data` objecs, i.e. you don't need to be
+             * a silent payments wallet, just someone interested in vending this
+             * data to light clients, e.g. a wallet service provider. In our
+             * example, Bob is scanning for himself but also sharing this data
+             * with light clients.
+             */
+            ret = secp256k1_silentpayments_recipient_public_data_create(ctx,
+                &public_data,
+                smallest_outpoint,
+                tx_input_ptrs, N_INPUTS,
+                NULL, 0 /* NULL because no eligible plain pubkey inputs were found in the tx */
+            );
+            assert(ret);
+            ret = secp256k1_silentpayments_recipient_public_data_serialize(ctx,
+                light_client_data33,
+                &public_data
+            );
+            assert(ret);
+
+            /* Scan the transaction */
+            n_found_outputs = 0;
+            ret = secp256k1_silentpayments_recipient_scan_outputs(ctx,
+                found_output_ptrs, &n_found_outputs,
+                tx_output_ptrs, N_OUTPUTS,
+                bob_scan_key,
+                &public_data,
+                &spend_pubkey,
+                label_lookup, &labels_cache /* NULL, NULL for no labels */
+            );
+            assert(n_found_outputs == 1);
+            printf("\n");
+            printf("Bob found the following outputs: \n");
+            for (i = 0; i < n_found_outputs; i++) {
+                printf("    ");
+                secp256k1_xonly_pubkey_serialize(ctx,
+                    xonly_print,
+                    &found_outputs[i].output
+                );
+                print_hex(xonly_print, sizeof(xonly_print));
+            }
+        }
+        {
+            /*** Scanning as a light client (Carol) ***
+             *
+             * Being a light client, Carol likely does not have access to the
+             * transaction outputs. This means she will need to first generate
+             * an output, check if it exists in the UTXO set (e.g. BIP158 or
+             * some other means of querying) and only proceed to check the next
+             * output (by incrementing `k`) if the first output exists.
```

Comment:
It seems that this description of the light client workflow would only work for users that don't use labels at all. While we could assume that indeed most light client users wouldn't use labels, I think we'd rather make a slighter more general example here assuming that Carol also use labels, or if we don't want to make the example more complicated at the very least explain it here in comments. Here's a quick rewriting of this paragraph based on how we made it work for Dana:
```
/** Being a light client, Carol likely does not have access to the
* transaction outputs. This means she will need to first generate
* the first outputs she would get in a transaction, meaning the output 
* from her plain spend_pubkey + one for each label she monitors 
* with _k_ == 0 , and check if it exists in the UTXO set (e.g. BIP158 or 
* some other means of querying).
*  
* As soon as she finds an output 
* she can request the whole transaction (or rather the whole block 
* to not reveal which transactions she's interested in) and use the
* same function than Bob from there to find any other output that 
* belongs to her.
*  
* She would then repeat this operation for each `public_data` she received
*/
```

Alternatively I guess she could also increment _k_ when she has a match and keep looking for outputs this way, but she needs to generate the _k_+1 output for each label also. Maybe that would indeed make more sense if we go for UTXO set scanning instead of scanning each block sequentially.

Reply:
Yes I agree it's kind of an edge case and we want to be careful not to compromise too much because of it, but on the other hand we should acknowledge it because it will definitely happen I think. I'll try to complete the tests on that part and see how bad it makes things